packages/runtime/test/transport/relationshipTemplates.test.ts

```diff
@@ -208,6 +209,99 @@ describe("Template Tests", () => {
+
+            const loadResult = await runtimeServices2.transport.relationshipTemplates.loadPeerRelationshipTemplate({
+                reference: createResult.value.truncatedReference,
+                password: "password"
+            });
+            expect(loadResult).toBeSuccessful();
+            expect(loadResult.value.password).toBe("password");
+        });
+
+        test("error when loading a template with a wrong password", async () => {
+            const createResult = await runtimeServices1.transport.relationshipTemplates.createOwnRelationshipTemplate({
+                content: emptyRelationshipTemplateContent,
+                expiresAt: DateTime.utc().plus({ minutes: 1 }).toString(),
+                password: "password"
+            });
+            expect(createResult).toBeSuccessful();
+
+            const loadResult = await runtimeServices2.transport.relationshipTemplates.loadPeerRelationshipTemplate({
+                reference: createResult.value.truncatedReference,
+                password: "wrong-password"
+            });
+            expect(loadResult).toBeAnError(/.*/, "error.platform.inputCannotBeParsed");
+        });
+
+        test("validation error when creating a template with a too short PIN", async () => {
+            const createResult = await runtimeServices1.transport.relationshipTemplates.createOwnRelationshipTemplate({
+                content: emptyRelationshipTemplateContent,
+                expiresAt: DateTime.utc().plus({ minutes: 1 }).toString(),
+                password: "1"
+            });
+            expect(createResult).toBeAnError(
+                "Your chosen 'password' is a PIN \\(consists of numbers only\\) and PINs must be at least 2 and at most 12 digits long",
+                "error.runtime.validation.invalidPropertyValue"
+            );
+        });
+
+        test("create a token for a password-protected template", async () => {
+            const createResult = await runtimeServices1.transport.relationshipTemplates.createOwnRelationshipTemplate({
+                content: emptyRelationshipTemplateContent,
+                expiresAt: DateTime.utc().plus({ minutes: 1 }).toString(),
+                password: "password"
+            });
+            expect(createResult).toBeSuccessful();
+            const createTokenResult = await runtimeServices1.transport.relationshipTemplates.createTokenForOwnTemplate({
+                templateId: createResult.value.id
+            });
+            expect(createTokenResult).toBeSuccessful();
+            const loadTokenResult = await runtimeServices2.transport.tokens.loadPeerToken({ reference: createTokenResult.value.truncatedReference, ephemeral: true });
+            expect(loadTokenResult).toBeSuccessful();
+            expect((loadTokenResult.value.content as TokenContentRelationshipTemplate).passwordType).toBe(1);
```

Comment:
```suggestion
            expect((loadTokenResult.value.content as TokenContentRelationshipTemplateJSON).passwordType).toBe(1);
```

please check that if you are using transport types that you are only using JSON representations. it's just luck that this typecast works.

Reply:
The JSON doesn't exist for this type and creating it for a key is iffy, I'll remove the typecast